src/ValidationUnit.js

```diff
@@ -185,6 +185,10 @@ export default class ValidationUnit {
     return this.setRequirement(requiredFunc, message, 'isRequired');
   }
 
+  isRequiredWhen(func, message = '{name} is required.') {
+    return this.setRequirement((val, allValues) => func(val, allValues) && requiredFunc(val, allValues) || !func(val, allValues), message, 'isRequired');
```

Comment:
I'm not totally sure about the order of operations here. Is it `(func(val, allValues) && requiredFunc(val, allValues)) || !func(val, allValues)`, or is it `func(val, allValues) && (requiredFunc(val, allValues) || !func(val, allValues))`. Maybe it would be better to include parens, just to be totally clear about the ordering you expect.